cmd/kpod/build.go

```diff
@@ -0,0 +1,143 @@
+package main
+
+import (
+	"os"
+	"os/exec"
+
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
+)
+
+var (
+	buildFlags = []cli.Flag{
+		cli.StringSliceFlag{
+			Name:  "build-arg",
+			Usage: "`argument=value` to supply to the builder",
+		},
+		cli.StringSliceFlag{
+			Name:  "file, f",
+			Usage: "`pathname or URL` of a Dockerfile",
+		},
+		cli.StringFlag{
+			Name:  "format",
+			Usage: "`format` of the built image's manifest and metadata",
+		},
+		cli.BoolTFlag{
+			Name:  "pull",
+			Usage: "pull the image if not present",
+		},
+		cli.BoolFlag{
+			Name:  "pull-always",
+			Usage: "pull the image, even if a version is present",
+		},
+		cli.BoolFlag{
+			Name:  "quiet, q",
+			Usage: "refrain from announcing build instructions and image read/write progress",
+		},
+		cli.StringFlag{
+			Name:  "runtime",
+			Usage: "`path` to an alternate runtime",
+		},
+		cli.StringSliceFlag{
+			Name:  "runtime-flag",
+			Usage: "add global flags for the container runtime",
+		},
+		cli.StringFlag{
+			Name:  "signature-policy",
+			Usage: "`pathname` of signature policy file (not usually used)",
+		},
+		cli.StringSliceFlag{
+			Name:  "tag, t",
+			Usage: "`tag` to apply to the built image",
+		},
+		cli.BoolFlag{
+			Name:  "tls-verify",
+			Usage: "Require HTTPS and verify certificates when accessing the registry",
+		},
+	}
```

Comment:
Docker build supports the following constraints, we need to plumb these into buildah/kpod build.
```

       -m, --memory=MEMORY
         Memory limit

       --memory-swap=LIMIT
          A limit value equal to memory plus swap. Must be used with the  -m (--memory) flag. The swap LIMIT should always be larger than -m (--memory) value.

       The format of LIMIT is <number>[<unit>]. Unit can be b (bytes), k (kilobytes), m (megabytes), or g (gigabytes). If you don't specify a unit, b is used. Set LIMIT to -1 to enable unlimited swap.

       --network=bridge
         Set the networking mode for the RUN instructions during build. Supported standard
         values are: bridge, host, none and container:<name|id>. Any other value
         is taken as a custom network's name or ID which this container should connect to.

       --shm-size=SHM-SIZE
         Size of /dev/shm. The format is <number><unit>. number must be greater than 0.
         Unit is optional and can be b (bytes), k (kilobytes), m (megabytes), or g (gigabytes). If you omit the unit, the system uses bytes.
         If you omit the size entirely, the system uses 64m.

       --cpu-shares=0
         CPU shares (relative weight).

       By default, all containers get the same proportion of CPU cycles.
         CPU shares is a 'relative weight', relative to the default setting of 1024.
         This default value is defined here:

                 cat /sys/fs/cgroup/cpu/cpu.shares
                 1024

       You can change this proportion by adjusting the container's CPU share
         weighting relative to the weighting of all other running containers.

       To modify the proportion from the default of 1024, use the --cpu-shares
         flag to set the weighting to 2 or higher.

                Container   CPU share    Flag
                {C0}        60% of CPU  --cpu-shares=614 (614 is 60% of 1024)
                {C1}        40% of CPU  --cpu-shares=410 (410 is 40% of 1024)

       The proportion is only applied when CPU-intensive processes are running.
         When tasks in one container are idle, the other containers can use the
         left-over CPU time. The actual amount of CPU time used varies depending on
         the number of containers running on the system.

       For example, consider three containers, where one has --cpu-shares=1024 and
         two others have --cpu-shares=512. When processes in all three
         containers attempt to use 100% of CPU, the first container would receive
         50% of the total CPU time. If you add a fourth container with --cpu-shares=1024,
         the first container only gets 33% of the CPU. The remaining containers
         receive 16.5%, 16.5% and 33% of the CPU.

                Container   CPU share   Flag                CPU time
                {C0}        100%        --cpu-shares=1024   33%
                {C1}        50%         --cpu-shares=512    16.5%
                {C2}        50%         --cpu-shares=512    16.5%
                {C4}        100%        --cpu-shares=1024   33%

       On a multi-core system, the shares of CPU time are distributed across the CPU
         cores. Even if a container is limited to less than 100% of CPU time, it can
         use 100% of each individual CPU core.

       For example, consider a system with more than three cores. If you start one
         container {C0} with --cpu-shares=512 running one process, and another container
         {C1} with --cpu-shares=1024 running two processes, this can result in the following
         division of CPU shares:

                PID    container    CPU    CPU share
                100    {C0}         0      100% of CPU0
                101    {C1}         1      100% of CPU1
                102    {C1}         2      100% of CPU2

       --cpu-period=0
         Limit the CPU CFS (Completely Fair Scheduler) period.

       Limit the container's CPU usage. This flag causes the kernel to restrict the
         container's CPU usage to the period you specify.

       --cpu-quota=0
         Limit the CPU CFS (Completely Fair Scheduler) quota.

       By default, containers run with the full CPU resource. This flag causes the kernel to restrict the container's CPU usage to the quota you specify.

       --cpuset-cpus=CPUSET-CPUS
         CPUs in which to allow execution (0-3, 0,1).

       --cpuset-mems=CPUSET-MEMS
         Memory nodes (MEMs) in which to allow execution (0-3, 0,1). Only effective on
         NUMA systems.

       For example, if you have four memory nodes on your system (0-3), use --cpuset-mems=0,1 to ensure the processes in your Docker container only use memory from the first two memory nodes.

       --cgroup-parent=CGROUP-PARENT
         Path to cgroups under which the container's cgroup are created.

       If the path is not absolute, the path is considered relative to the cgroups path of the init process.  Cgroups are created if they do not already exist.

       --ulimit=[]
         Ulimit options

       +-v, --volume=[] Create a bind mount +   (format: host-dir:container-dir[:<suffix options>], where suffix options +are comma delimited and selected from [ro] and [z|Z].). Read-write mode isn't
       supported at build time.  +In case rw is specified a warning is printed during the build and it will be changed to ro preserving any SELinux mode provided.  +

       For more information about ulimit see Setting ulimits in a container ⟨https://docs.docker.com/engine/reference/commandline/run/#set-ulimits-in-container---ulimit⟩

```

Reply:
Yes, but for this PR, I'd like to concentrate on just matching Buildah, then circling back.

Reply:
Ok